### PR TITLE
Added Auth0 scaffolding and rules for safekeeping.

### PR DESCRIPTION
**Pull request by jacobdavis1**
# housingxyz :: pull-request

## change proposal

- [x] Interceptor
- [x] Auth Guard
- [x] Auth Service
- [x] Removed some MSAL references; still wading through them
- [x] Modified Module and Router files to reflect new auth
## reviewer

- @escalonn

## contributing

this project is maintained under the terms of the [Contribution Guidelines][contribution-guidelines-url].

[contribution-guidelines-url]: https://github.com/housingxyz/housingxyz/blob/master/.github/CONTRIBUTING.md 'CONTRIBUTION GUIDELINES'

### CODE REVIEW

File: housing/auth0rules.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Add permissions to tokens (turned off unless really needed)
```
Comment (escalonn) [inline housing/auth0rules.js]: document purpose of having this file in repo

File: housing/package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "@angular/platform-browser": "~8.2.7",
     "@angular/platform-browser-dynamic": "~8.2.7",
     "@angular/router": "~8.2.7",
+    "@auth0/auth0-spa-js": "^1.5.0",
     "@azure/msal-angular": "^0.1.4",
```
Comment (escalonn) [inline housing/package.json]: consider yanking this out

File: housing/src/app/app.component.ts [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+export class AppComponent implements OnInit {
   title = 'housing';
+
+  constructor(private auth: AuthService) {}
+
+  ngOnInit() {
+    this.auth.localAuthSetup();
+    this.auth.handleAuthCallback();
+
+    this.auth.getUser$().subscribe( res => console.log(res));
```
Comment (escalonn) [inline housing/src/app/app.component.ts]: probably not `console.log` in master branch
also, extra space

File: housing/src/app/services/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import Auth0Client from '@auth0/auth0-spa-js/dist/typings/Auth0Client';
+import { from, of, Observable, BehaviorSubject, combineLatest, throwError } from 'rxjs';
+import { tap, catchError, concatMap, shareReplay } from 'rxjs/operators';
+import { Router } from '@angular/router';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
```
Comment (escalonn) [inline housing/src/app/services/auth.service.ts]: document where this code was taken from
and _any_ manual modifications to that

File: housing/src/app/services/interceptor.service.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  HttpInterceptor
+} from '@angular/common/http';
+import { AuthService } from './auth.service';
+import { Observable, throwError } from 'rxjs';
+import { mergeMap, catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class InterceptorService implements HttpInterceptor {
```
Comment (escalonn) [inline housing/src/app/services/interceptor.service.ts]: same thing, document source, for all such

File: housing/package.json (hunk 2 above)
Comment (jacobdavis1) [reply housing/package.json]: Will leave it in until MSAL can be totally eradicated.

**Pull request merged**